ulib/FStar.Tactics.Derived.fst

```diff
@@ -599,12 +599,12 @@ let unfold_def (t:term) : Tac unit =
 equalities. The lemmas need to prove *propositional* equalities, that
 is, using [==]. *)
 let l_to_r (lems:list term) : Tac unit =
-    let first_or_trefl () : Tac unit =
+    let first_or_skip () : Tac unit =
```

Comment:
A few comments about this change:
IIUC, rather than applying trefl in case all the rewriting lemmas fail, this change will instead fail silently. 

1. I don't see changes anywhere to any callers of l_to_r. How come? 
2. Because of the fold_left, if I write `l_to_r [l1;l2;l3]` wouldn't this actually try to apply 

```
(fun () -> apply_lemma_r1 l3) `or_else`
(fun () -> apply_lemma_r1 l2) `or_else`
(fun () -> apply_lemma_r1 l1) `or_else`
(fun () -> fail_silently "SKIP")
```

i.e., isn't this backwards? Or rather isn't it more like "last_or_skip". Looks like that was the original behavior too ... is it intentional?

Reply:
Re: 1, the implementation of `ctrl_rewrite` hardcodes a special behaviour for a failure with Msg "SKIP", where it simply rolls back the current proofstate to before generating the equality, hence avoiding unresolved uvars when all rewritings fail.

Re: 2, you're indeed right, testing it with the small program below, the last lemma is indeed the one called. Another slightly annoying thing is that it stops at the first lemma it can apply instead of trying them all if rewritings can be chained. I think all our usecases so far use `l_to_r` with only one lemma, but it'd be good to change in the future.
```fstar
assume val f : int
assume val g : int
assume val h : int

assume
val lem1 (u:unit) : Lemma (f == g)

assume
val lem2 (u:unit) : Lemma (f == h)

open FStar.Tactics

let test () =
  assert (f == h) by (l_to_r [(`lem1); (`lem2)]; dump "here")
```